src/model/SurfacePropertyExposedFoundationPerimeter.hpp

```diff
@@ -70,6 +70,8 @@ namespace model {
 
     std::string surfaceName() const;
```

Comment:
I think you want `OS_DEPRECATED` for that one. Remove from Impl, and do `surface().nameString()` in the deprecated method